apiserver/facades/controller/charmrevisionupdater/updater.go

```diff
@@ -37,11 +37,12 @@ type CharmRevisionUpdaterAPI struct {
 	store objectstore.ObjectStore
 	clock clock.Clock
 
-	newCharmhubClient newCharmhubClientFunc
-	logger            corelogger.Logger
+	modelConfigService ModelConfigService
+	newCharmhubClient  newCharmhubClientFunc
+	logger             corelogger.Logger
 }
 
-type newCharmhubClientFunc func(st State) (CharmhubRefreshClient, error)
+type newCharmhubClientFunc func(context.Context) (CharmhubRefreshClient, error)
```

Comment:
Because the New function for creating the API is publicly expose in this package I would say this type should be a publicly exported as it is a dependency type.

Reply:
done